Mathlib/Topology/LocallyConstant/Algebra.lean

```diff
@@ -291,48 +291,118 @@ theorem coe_algebraMap (r : R) : ⇑(algebraMap R (LocallyConstant X Y) r) = alg
 
 end Algebra
 
+section Eval
+
+/-- Evaluation as a `MonoidHom` -/
+@[to_additive (attr := simps) "Evaluation as an `AddMonoidHom`"]
+def evalMonoidHom [MulOneClass Z] (x : X) : LocallyConstant X Z →* Z where
+  toFun f := f x
+  map_mul' _ _ := by simp only [LocallyConstant.coe_mul, Pi.mul_apply]
+  map_one' := rfl
```

Comment:
```suggestion
def evalMonoidHom [MulOneClass Z] (x : X) : LocallyConstant X Z →* Z :=
  (Pi.evalMonoidHom x).comp coeFnMonoidHom
```
or similar, same throughout

Reply:
I made these changes, but had to define more algebraic versions of `coeFn` to do so. There was also no linear version of `Pi.eval`, and since the file `Mathlib/Algebra/Module/Pi` doesn't know about linear maps, I added it to `Mathlib/Algebra/Module/LinearMap` (`#find_home` agrees with this location). Is this what you had in mind?